api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
If we do the `parser.startFile()` thing, then I suppose a Parser could throw an error if it has `hasHeaderRow` / `skipHeaderRows` set, and has its `parse()` method called without an earlier `startFile()` call. Since that would mean it's being used in "stream mode" with parameters that only make sense for files.